src/components/Dialogs/Conditions/ConditionsManager.js

```diff
@@ -27,22 +47,40 @@ const ConditionsManager = ({
     setConditionsByDimension,
 }) => {
     const [conditionsList, setConditionsList] = useState(
-        parseConditionsStringToArray(conditions)
+        (conditions.condition?.length &&
+            parseConditionsStringToArray(conditions.condition)) ||
+            (conditions.legendSet ? [EMPTY_CONDITION] : [])
     )
 
-    const addCondition = () => setConditionsList([...conditionsList, ''])
+    const [selectedLegendSet, setSelectedLegendSet] = useState(
+        conditions.legendSet
+    )
 
-    const removeCondition = (conditionIndex) =>
-        setConditionsList(
-            conditionsList.filter((_, index) => index !== conditionIndex)
+    const addCondition = () =>
+        setConditionsList([...conditionsList, EMPTY_CONDITION])
+
+    const removeCondition = (conditionIndex) => {
+        const filteredConditionsList = conditionsList.filter(
+            (_, index) => index !== conditionIndex
         )
+        setConditionsList(filteredConditionsList)
+        if (
+            selectedLegendSet &&
+            !filteredConditionsList.some((condition) =>
+                condition.includes(OPERATOR_RANGE_SET)
+            )
+        ) {
+            setSelectedLegendSet(null)
+        }
+    }
 
-    const setCondition = (conditionIndex, value) =>
+    const setCondition = (conditionIndex, value) => {
         setConditionsList(
             conditionsList.map((condition, index) =>
```

Comment:
Yeah, but since you need to pass it in as a value to `setConditionsList` (and can't operate on the list directly, you'd have to clone the list, change the value for the index, then pass the new list to the fn. This essentially does this but on one line. As far as looping through vs cloning goes, I suppose it'll be rare to see more than 10 conditions used at once, so from a performance perspective it should be similar?

Reply:
Yeah, I had a similar problem recently and ended up copying the array and using splice, in that case I also don't expect hundreds of elements in the list so it probably doesn't make much difference.
👍